src/MultiOwnerLightAccount.sol

```diff
@@ -0,0 +1,336 @@
+        return _ENTRY_POINT;
+    }
+
+    ///@notice Return the owners of this account.
+    ///@return The array of owner addresses.
+    function owners() public view returns (address[] memory) {
+        return _getStorage().owners.getAll().toAddressArray();
+    }
+
+    /// @notice Check current account deposit in the entry point.
+    /// @return The current account deposit.
+    function getDeposit() public view returns (uint256) {
+        return entryPoint().balanceOf(address(this));
+    }
+
+    /// @notice Returns the domain separator for this contract, as defined in the EIP-712 standard.
+    /// @return bytes32 The domain separator hash.
+    function domainSeparator() public view returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                _DOMAIN_SEPARATOR_TYPEHASH,
+                _NAME_HASH, // name
+                _VERSION_HASH, // version
+                block.chainid, // chainId
+                address(this) // verifying contract
+            )
+        );
+    }
+
+    /// @notice Returns the pre-image of the message hash.
+    /// @param message Message that should be encoded.
+    /// @return Encoded message.
+    function encodeMessageData(bytes memory message) public view returns (bytes memory) {
+        bytes32 messageHash = keccak256(abi.encode(_LA_MSG_TYPEHASH, keccak256(message)));
+        return abi.encodePacked("\x19\x01", domainSeparator(), messageHash);
+    }
+
+    /// @notice Returns hash of a message that can be signed by owners.
+    /// @param message Message that should be hashed.
+    /// @return Message hash.
+    function getMessageHash(bytes memory message) public view returns (bytes32) {
+        return keccak256(encodeMessageData(message));
+    }
+
+    /// @dev The signature is valid if it is signed by the owner's private key (if the owner is an EOA) or if it is a
+    /// valid ERC-1271 signature from the owner (if the owner is a contract). Note that unlike the signature validation
+    /// used in `validateUserOp`, this does **not** wrap the digest in an "Ethereum Signed Message" envelope before
+    /// checking the signature in the EOA-owner case.
+    /// @inheritdoc IERC1271
+    function isValidSignature(bytes32 digest, bytes memory signature) public view override returns (bytes4) {
+        bytes32 messageHash = getMessageHash(abi.encode(digest));
+        (address recovered, ECDSA.RecoverError error,) = messageHash.tryRecover(signature);
+        if (error == ECDSA.RecoverError.NoError && _getStorage().owners.contains(CastLib.toSetValue(recovered))) {
+            return _1271_MAGIC_VALUE;
+        }
+        if (_isValidERC1271SignatureNow(messageHash, signature)) {
+            return _1271_MAGIC_VALUE;
+        }
+        return 0xffffffff;
+    }
+
+    function _initialize(address[] calldata owners_) internal virtual {
+        emit LightAccountInitialized(_ENTRY_POINT, owners_);
+        _updateOwners(owners_, new address[](0));
```

Comment:
Can we replace this with `_addOwnersOrRevert`? It would remove the `OwnersUpdated` event, but this data is present already in the `LightAccountInitialized` event.

Reply:
We emit `OwnershipTransferred` for LightAccount on initialize, so this was an attempt to make things consistent there. Shall we remove it there as well?

Reply:
ah... from an integration standpoint, it's probably easier to just index the one event `OwnershipTransferred`. Arguably the `LightAccountInitialized` event isn't really needed, but I guess it's fine to keep as-is and have some small duplcation.